services/transactionService.ts

```diff
@@ -165,43 +115,23 @@ export async function upsertManyTransactionsForAddress (transactions: Transactio
 
 export async function syncTransactionsForAddress (addressString: string): Promise<TransactionWithAddressAndPrices[]> {
   const address = await fetchAddressBySubstring(addressString)
-  let newTransactionsCount = -1
-  let seenTransactionsCount = 0
-  let insertedTransactions: TransactionWithAddressAndPrices[] = []
-  while (newTransactionsCount !== 0) {
-    const addressTransactions = await getAddress({
-      address: address.address,
-      nbFetch: FETCH_N,
-      nbSkip: seenTransactionsCount
-    })
-    let nextConfirmedTransactions = addressTransactions.confirmedTransactionsList
-    let nextUnconfirmedTransactions = addressTransactions.unconfirmedTransactionsList.map(mempoolTx => parseMempoolTx(mempoolTx))
-
-    // remove transactions older than the networks
-    nextConfirmedTransactions = nextConfirmedTransactions.filter(txThesholdFilter(address))
-    nextUnconfirmedTransactions = nextUnconfirmedTransactions.filter(txThesholdFilter(address))
 
-    newTransactionsCount = nextConfirmedTransactions.length // Unconfirmed transactions are NOT taken into account when skipping with nbSkip
-    seenTransactionsCount += newTransactionsCount
-
-    let newInsertedTransactions = await upsertManyTransactionsForAddress(nextConfirmedTransactions, address)
-    newInsertedTransactions = [
-      ...newInsertedTransactions,
-      ...(await upsertManyTransactionsForAddress(nextUnconfirmedTransactions, address, false))
-    ]
-    insertedTransactions = [...insertedTransactions, ...newInsertedTransactions]
-    await new Promise(resolve => setTimeout(resolve, FETCH_DELAY))
-  }
+  const addressTransfers = await getAddressTransfers(addressString)
```

Comment:
I don't like this refactor at all:
1. You are defining now this `getAddressTransfers` in two different places:  `grpcService` and `chronikService`, such that most of it is duplicated code;
2. But the bigger problem is that: notice that before the function would go into upserting transactions as they were fetched from the thrid party; for each fetched page we would insert the transactions into the database before getting the next `FETCH_N` transactions. Now it's just going to await until it has fetched ALL the transactions, page by page, and then it will upsert into the database. There are addresses with 20k+ transactions which will therefore never complete or sync anything. This also makes impossible to implement #403 point 3.

Reply:
I will consider that when creating the smaller incremental PRs